src/utils/authorizeDelegatedAccount.ts

```diff
@@ -0,0 +1,55 @@
+import { Contract, Wallet, providers } from 'ethers'
+import * as DelegatedAccessRegistry from '../contracts/DelegatedAccessRegistry.sol/DelegatedAccessRegistry.json'
+
+import EthCrypto from 'eth-crypto'
+import handleError from '$/utils/handleError'
+import { Address } from '$/types'
+
+const DelegatedAccessRegistryAddress = '0xf5803cdA6352c515Ee11256EAA547BE8422cC4EE'
+
+export const signDelegatedChallenge = async (
+    metamaskAddress: string,
+    delegatedPrivateKey: string,
+    challengeType: 0 | 1 // 0 = authorize | 1 = revoke
+) => {
+    const delegated = new Wallet(delegatedPrivateKey)
+
+    const message = EthCrypto.hash.keccak256([
+        { type: 'uint256', value: challengeType.toString() },
+        { type: 'address', value: metamaskAddress },
+    ])
+
+    const signature = EthCrypto.sign(delegated.privateKey, message)
```

Comment:
It can also simply be

```ts
const signature = sign(delegatedPrivateKey, message)
```
Meaning, there's no need for `new Wallet(delegatedPrivateKey)` line.